frontend/ui/uimanager.lua

```diff
@@ -34,10 +34,15 @@ local UIManager = {
 function UIManager:init()
     self.event_handlers = {
         __default__ = function(input_event)
-            self:sendEvent(input_event)
+            -- Suspension in Kobo can be interrupted by screen updates. We
+            -- ignore user touch input here so screen udpate won't be
+            -- triggered in suspend mode
+            if not Device.screen_saver_mode then
+                self:sendEvent(input_event)
+            end
```

Comment:
I cannot quite tell what's the "bricked" situation you have mentioned. I just imagined this looks like a common logic which is reasonable across different platforms.
I ran this logic on my Kindle for several days, and it looks normal. I believe Kindle has its own power management subsystem, which we cannot easily intercept. So using this logic should not impact it.
To ensure it's safe, I will revert this change if it was for Kobo only.